app/scripts/modules/amazon/src/loadBalancer/configure/network/NLBListeners.tsx

```diff
@@ -43,23 +68,84 @@ export class NLBListeners
     return errors;
   }
 
+  private needsCert(listener: IListenerDescription): boolean {
+    return listener.protocol === 'TLS';
+  }
+
+  private showCertificateSelect(certificate: IALBListenerCertificate): boolean {
+    return certificate.type === 'iam' && this.state.certificates && Object.keys(this.state.certificates).length > 0;
+  }
+
+  public componentDidMount(): void {
+    this.loadCertificates();
+  }
+
+  private loadCertificates(): void {
+    AmazonCertificateReader.listCertificates().then((certificates) => {
+      this.setState({ certificates });
+    });
+  }
+
   private updateListeners(): void {
     this.props.formik.setFieldValue('listeners', this.props.formik.values.listeners);
   }
 
+  private addListenerCertificate(listener: IListenerDescription): void {
+    listener.certificates = listener.certificates || [];
+    listener.certificates.push({
+      certificateArn: undefined,
+      type: 'iam',
+      name: undefined,
+    });
```

Comment:
```suggestion
    (listener.certificates ?? []).push({
      certificateArn: undefined,
      type: 'iam',
      name: undefined,
    });
```

Reply:
no, this will push onto the `[]` which isn't saved anywhere

Reply:
never mind then I didn’t realize that. I’ll delete this suggestion